File: 287-Find-the-Duplicate-Number/py/main.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hasSeenNumber = [0] * len(nums)
        for num in nums:
            if hasSeenNumber[num] == -1:
                return num
            hasSeenNumber[num] = -1
        return -1

File: 287-Find-the-Duplicate-Number/rs/main.rs
fm main() {
	println!("Hello workd");
}

pub fn find_duplicate(nums: Vec<i32>) -> i32 {
        let mut hasSeenNumber = nums.clone();
        for &num in nums.iter() {
            if hasSeenNumber[num as usize] == -1 {
                return num;
            }
            hasSeenNumber[num as usize] = -1;
        }
        return -1;
}
